packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -436,6 +444,16 @@ class SegmentedButton<T> extends StatefulWidget {
 @visibleForTesting
 class SegmentedButtonState<T> extends State<SegmentedButton<T>> {
   bool get _enabled => widget.onSelectionChanged != null;
+  bool _hovering = false;
+  bool _focused = false;
+  bool get _selected => widget.selected.contains(widget.segments.first.value);
```

Comment:
This doesn't look right. I think if any of the segments are selected, we can consider the SegmentedButton as selected, not only for the first segement.

Reply:
Fixed